core/metrics-core-service/src/test/java/org/hawkular/metrics/core/jobs/DeleteTenantITest.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.jobs;
+
+import static java.util.Arrays.asList;
+
+import static org.hawkular.metrics.core.service.Order.ASC;
+import static org.hawkular.metrics.datetime.DateTimeService.currentMinute;
+import static org.hawkular.metrics.model.AvailabilityType.DOWN;
+import static org.hawkular.metrics.model.AvailabilityType.UP;
+import static org.hawkular.metrics.model.MetricType.AVAILABILITY;
+import static org.hawkular.metrics.model.MetricType.COUNTER;
+import static org.hawkular.metrics.model.MetricType.GAUGE;
+import static org.hawkular.metrics.model.MetricType.STRING;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.hawkular.metrics.core.service.BaseITest;
+import org.hawkular.metrics.core.service.DataAccess;
+import org.hawkular.metrics.core.service.DataAccessImpl;
+import org.hawkular.metrics.core.service.MetricsServiceImpl;
+import org.hawkular.metrics.datetime.DateTimeService;
+import org.hawkular.metrics.model.AvailabilityType;
+import org.hawkular.metrics.model.DataPoint;
+import org.hawkular.metrics.model.Metric;
+import org.hawkular.metrics.model.MetricId;
+import org.hawkular.metrics.model.MetricType;
+import org.hawkular.metrics.model.Tenant;
+import org.hawkular.metrics.scheduler.api.JobDetails;
+import org.hawkular.metrics.scheduler.impl.SchedulerImpl;
+import org.hawkular.metrics.sysconfig.ConfigurationService;
+import org.joda.time.DateTime;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.codahale.metrics.MetricRegistry;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.google.common.collect.ImmutableMap;
+
+import rx.Observable;
+import rx.Subscription;
+import rx.functions.Action1;
+import rx.schedulers.Schedulers;
+import rx.schedulers.TestScheduler;
+import rx.subjects.PublishSubject;
+
+/**
+ * @author jsanda
+ */
+public class DeleteTenantITest extends BaseITest {
```

Comment:
Sure, those would be good tests to include. I also want to point that the job is idempotent (as most data access operations should be in Cassandra to the extent possible). Deleting a non-existent tenant should basically be a no-op. Deleting the same tenant twice would result in duplicate work, but because the job is idempotent, nothing would break. With that said, I am thinking it might be worthwhile to add some checks that will avoid submitting a duplicate job to avoid the extra work.